bolts/common.py

```diff
@@ -1,7 +1,31 @@
 from __future__ import annotations
+import collections
+from typing import Generic
 
-from typing_extensions import Literal
+from typing_extensions import ClassVar, Literal
 
 __all__ = ["Stage"]
 
+from typing_inspect import typingGenericAlias
```

Comment:
wait, is `typing_inspect` a dependency? then you can just import `get_args` from there. I copied it from there.

Reply:
`ethicml > typed-argument-parser > typing-inspect`

Reply:
I see. Maybe we should get rid of that dependency on typed-argument-parser.

In any case, I don't think it makes sense to do a mix of copying from typing-inspect and using typing-inspect as a library.

Reply:
I'll use `typing-inspect` until we get rid of it from ethicml.